Packs/Base/Scripts/ValidateContent/ValidateContent.py

```diff
@@ -0,0 +1,384 @@
+    lint_log_dir = os.path.dirname(json_output_file)
+    output_capture = io.StringIO()
+    with redirect_stdout(output_capture):
+        with redirect_stderr(output_capture):
+            lint_manager = LintManager(
+                input=file_path, git=False, all_packs=False, quiet=False, verbose=1,
+                log_path=lint_log_dir, prev_ver='', json_file_path=json_output_file
+            )
+            lint_manager.run_dev_packages(
+                parallel=1, no_flake8=False, no_xsoar_linter=False, no_bandit=False, no_mypy=False,
+                no_vulture=False, keep_container=False, no_pylint=True, no_test=True, no_pwsh_analyze=True,
+                no_pwsh_test=True, test_xml='', failure_report=lint_log_dir
+            )
+
+
+def run_parallel_operations(operations: List[Tuple[Callable, List]]) -> None:
+    parallel_ops = len(operations)
+    with concurrent.futures.ThreadPoolExecutor(max_workers=parallel_ops) as executor:
+        try:
+            for func_to_execute, func_args in operations:
+                executor.submit(func_to_execute, *func_args)
+        except Exception as e:
+            demisto.debug(f'Stopping concurrent operations of validation script due to error - {e}')
+            try:
+                executor.shutdown(wait=False)
+            except Exception as e:
+                demisto.debug(f'Failed shutting down executor - {e}')
+            raise
+
+
+def validate_content(filename, data, tmp_directory: str) -> List:
+    # case when a pack zip file has been passed
+    content = Content(tmp_directory)
+    json_output_path = os.path.join(tmp_directory, 'validation_res.json')
+    lint_output_path = os.path.join(tmp_directory, 'lint_res.json')
+    if filename.endswith('.zip'):
+        # write zip file data to file system
+        zip_path = os.path.abspath(os.path.join(tmp_directory, filename))
+        with open(zip_path, 'wb') as fp:
+            fp.write(data)
+
+        pack_name = get_pack_name(zip_path)
+        contrib_converter = ContributionConverter(name=pack_name, contribution=zip_path, base_dir=tmp_directory)
+        convert_contribution_to_pack(contrib_converter)
+        # Call the standalone function and get the raw response
+        os.remove(zip_path)
+        pack_path = contrib_converter.pack_dir_path
+        pack_entity = Pack(pack_path)
+        for content_entities in pack_entity.scripts, pack_entity.integrations:
+            for content_entity in content_entities:
+                buff = io.StringIO()
+                yaml.dump(content_entity.to_dict(), buff)
+                content_entity.path.write_text(buff.getvalue())
+        path_to_validate = pack_path
+    else:
+        # a single content item
+        pack_name = 'TmpPack'
+        pack_dir = content.path / 'Packs' / pack_name
+        # create pack_metadata.json file in TmpPack
+        contrib_converter = ContributionConverter(name=pack_name, base_dir=tmp_directory, pack_dir_name=pack_name)
```

Comment:
using ContributionConverater means it converts the file/pack to be formatted like `demisto-sdk format`?

Reply:
no
if you remember we decided not to run format here, at least for start, as it wont actually modify the files and by that wont return errors to the user